receiver/awscontainerinsightreceiver/internal/stores/nodeinfo.go

```diff
@@ -0,0 +1,97 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stores
+
+import (
+	"log"
+	"sync"
+)
+
+type nodeStats struct {
+	podCnt       int
+	containerCnt int
+	cpuReq       int64
+	memReq       int64
+}
+
+type nodeInfo struct {
+	nodeStats nodeStats
+	cpuLock   *sync.RWMutex
+	memLock   *sync.RWMutex
```

Comment:
Why are these `*sync.RWMutex` and not `sync.RWMutex`?  It seems odd to use a pointer to a mutex.